integrations/gcp/gcp_core/helpers/ratelimiter/base.py

```diff
@@ -0,0 +1,94 @@
+from abc import ABC, abstractmethod
+from typing import Any, Optional, TYPE_CHECKING
+from aiolimiter import AsyncLimiter
+from google.cloud.cloudquotas_v1 import CloudQuotasAsyncClient, GetQuotaInfoRequest  # type: ignore
+from loguru import logger
+from enum import Enum
+from port_ocean.context.ocean import ocean
+from gcp_core.cache import cache_coroutine_result  # type: ignore
+from collections.abc import MutableSequence
+
+
+_DEFAULT_RATE_LIMIT_TIME_PERIOD: float = 60.0
+_DEFAULT_RATE_LIMIT_QUOTA: int = int(
+    ocean.integration_config["search_all_resources_per_minute_quota"]
+)
+
+if TYPE_CHECKING:
+    from google.cloud.cloudquotas_v1 import DimensionsInfo
+
+
+class ContainerType(Enum):
+    PROJECT = "projects"
+    FOLDER = "folders"
+    ORGANIZATION = "organizations"
+
+
+class GCPResourceQuota(ABC):
+    quota_id: str | None = None
+    service: str | None = None
+    container_type: ContainerType = ContainerType.PROJECT
+    _default_quota: int = _DEFAULT_RATE_LIMIT_QUOTA
+
```

Comment:
add comment about this class usage and why we need it and why we chose to implement it in this way